src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp

```diff
@@ -0,0 +1,232 @@
+  pose_name_parameter.description.name = "target_state_name";
+  pose_name_parameter.description.type = BehaviorParameterDescription::TYPE_STRING;
+  pose_name_parameter.string_value = "Home";
+
+  do_objective_goal->parameter_overrides.push_back(pose_name_parameter);
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal)));
+}
+
+TEST_F(ObjectiveFixture, TestMoveToWaypoint)
+{
+  ASSERT_TRUE(setupDoObjectiveSequenceClient());
+
+  auto do_objective_goal = std::make_unique<DoObjectiveSequence::Goal>();
+  do_objective_goal->objective_name = "Move to Waypoint";
+
+  BehaviorParameter waypoint_name_parameter;
+  waypoint_name_parameter.behavior_namespaces.emplace_back("move_to_waypoint");
+  waypoint_name_parameter.description.name = "waypoint_name";
+  waypoint_name_parameter.description.type = BehaviorParameterDescription::TYPE_STRING;
+  waypoint_name_parameter.string_value = "Forward Down";
+
+  do_objective_goal->parameter_overrides.push_back(waypoint_name_parameter);
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal)));
+}
+
+TEST_F(ObjectiveFixture, TestCloseGripper)
+{
+  ASSERT_TRUE(setupDoObjectiveSequenceClient());
+
+  auto do_objective_goal = std::make_unique<DoObjectiveSequence::Goal>();
+  do_objective_goal->objective_name = "Close Gripper";
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal)));
+}
+
+TEST_F(ObjectiveFixture, TestOpenGripper)
+{
+  ASSERT_TRUE(setupDoObjectiveSequenceClient());
+
+  auto do_objective_goal = std::make_unique<DoObjectiveSequence::Goal>();
+  do_objective_goal->objective_name = "Open Gripper";
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal)));
+}
+
+TEST_F(ObjectiveFixture, TestPickObject)
+{
+  ASSERT_TRUE(setupDoObjectiveSequenceClient());
+
+  auto do_objective_goal = std::make_unique<DoObjectiveSequence::Goal>();
+  do_objective_goal->objective_name = "Pick Object";
+
+  // Set a grasp pose based on a user-selected grasp originally picked in simulation
+  const auto grasp_pose = [] {
+    PoseStamped msg;
+    msg.header.frame_id = "world";
+    msg.pose.position = geometry_msgs::build<Point>().x(0.58976525).y(-0.1367085).z(0.27501845);
+    msg.pose.orientation = geometry_msgs::build<Quaternion>().x(0.7063694).y(0.44723025).z(0.29349735).w(0.46355873);
+    return msg;
+  }();
+
+  // Update the grasp pose parameter used when planning the objective
+  BehaviorParameter pose_name_parameter;
+  pose_name_parameter.behavior_namespaces.emplace_back("pick_object");
+  pose_name_parameter.description.name = "grasp_pose";
+  pose_name_parameter.description.type = BehaviorParameterDescription::TYPE_POSE;
+  pose_name_parameter.pose_value = grasp_pose;
+
+  do_objective_goal->parameter_overrides.insert(do_objective_goal->parameter_overrides.end(), { pose_name_parameter });
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal), 30.0));
+}
+
+TEST_F(ObjectiveFixture, DISABLED_TestInspectSurface)
```

Comment:
Does this test work if we enable it?

If not, can we add a comment for why?

Reply:
https://github.com/PickNikRobotics/moveit_studio/pull/1111

Reply:
```
| 1: [test_basic_ur5e_objectives-19] [ RUN      ] ObjectiveFixture.TestInspectSurface
| 1: [test_basic_ur5e_objectives-19] 1703219439.154323 [49] test_basic: selected interface "lo" is not multicast-capable: disabling multicast
| 1: [test_basic_ur5e_objectives-19] /home/alex/workspace/moveit_studio_ws/src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:141: Failure
| 1: [test_basic_ur5e_objectives-19] Expected equality of these values:
| 1: [test_basic_ur5e_objectives-19]   set_transform_future.wait_for(std::chrono::seconds(10))
| 1: [test_basic_ur5e_objectives-19]     Which is: 4-byte object <01-00 00-00>
| 1: [test_basic_ur5e_objectives-19]   std::future_status::ready
| 1: [test_basic_ur5e_objectives-19]     Which is: 4-byte object <00-00 00-00>
| 1: [test_basic_ur5e_objectives-19] [  FAILED  ] ObjectiveFixture.TestInspectSurface (10238 ms)

```

Reply:
I fixed all your other comments and will leave this disabled.

There are 3 more test suites which were also suppose to be ported but I didn't because they were disabled I will continue working on those or create issues to do them individually and include this then.